clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -2533,10 +2534,7 @@ public void testInflightBatchesExpireOnDeliveryTimeout() throws InterruptedExcep
         time.sleep(deliveryTimeoutMs);
         sender.runOnce();  // receive first response
         assertEquals(0, sender.inFlightBatches(tp0).size(), "Expect zero in-flight batch in accumulator");
-        assertInstanceOf(
-            TimeoutException.class,
-            assertThrows(ExecutionException.class, request::get).getCause(),
-            "The expired batch should throw a TimeoutException");
+        assertDoesNotThrow(() -> request.get());
```

Comment:
Instead of a new integration test, I am wondering if we could just extend this test for coverage. We can create an inflight request, but not complete the request. Let the delivery timeout to expire and check that the batch is not completed.